### PR TITLE
Update index-state to 2020-10-01

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: cabal.project
Comment (mrBliss) [inline cabal.project:143]: Merge these to and use two `subdir`s, like in the rest of the file :slightly_smiling_face:

File: cabal.project
Comment (mrBliss) [inline cabal.project:174]: Is this one needed already? I was only going to add it when we really use it (#2679).

File: cabal.project
Comment (mrBliss) [inline cabal.project:161]: Any breaking changes?

**Comment by mrBliss** [comment]
@newhoggy I'm curious why you are pushing this change. `cardano-node` already uses the new index-state. Is it just for consistency?

It's just that it's not ideal when people unrelated to the repo or dependency, update a dependency that might cause lots of breaking changes (see the CI failures). I'm currently working on a dependency update of `cardano-ledger-specs` in #2679, but first three more PRs (https://github.com/input-output-hk/cardano-ledger-specs/pull/1922, https://github.com/input-output-hk/cardano-ledger-specs/pull/1902, https://github.com/input-output-hk/cardano-ledger-specs/pull/1923) need to be merged in `cardano-ledger-specs`.

Unless you can drop the dependency bump of `cardano-ledger-specs`, this PR will have to wait until the four PRs I mentioned are merged.

File: cabal.project
Comment (newhoggy) [reply cabal.project:174]: I only added it because the build failed.

**Comment by newhoggy** [comment]
Thanks for your advice.  Fortunately, I the earlier decision to make the new version of `cardano-prelude` backwards compatible means I ought to be able to do as you suggest and not bump `cardano-ledger-spec`.  I'll try that now and see how it goes.

**Review by mrBliss — APPROVED** [review]

**Comment by mrBliss** [comment]
bors merge

**Pull request merged**
